src/be/mcjava/model/CustomerOrder.java

```diff
@@ -0,0 +1,89 @@
+package be.mcjava.model;
+
+
+import java.util.List;
+
+/**
+ * Contains all data related to the order of a customer
+ */
+public class CustomerOrder {
+    private Long id;
+    private String name;
+    private String telephoneNumber;
+    private List<OrderItem> itemsToOrder;
+    private boolean finished;
+    
+    public CustomerOrder(String name, String telephoneNumber, List<OrderItem> itemsToOrder) {
+        this(null,name,telephoneNumber,itemsToOrder);
+    }
+    
+    public CustomerOrder(Long id, String name, String telephoneNumber, List<OrderItem> itemsToOrder) {
+        setId(id);
+        setName(name);
+        setTelephoneNumber(telephoneNumber);
+        setItemsToOrder(itemsToOrder);
+    }
+    
+    public Long getId() {
+        return id;
+    }
+    
+    public void setId(Long id) {
+        this.id = id;
+    }
+    
+    public String getName() {
+        return name;
+    }
+    
+    public void setName(String name) {
+        this.name = name;
+    }
+    
+    public String getTelephoneNumber() {
+        return telephoneNumber;
+    }
+    
+    public void setTelephoneNumber(String telephoneNumber) {
+        this.telephoneNumber = telephoneNumber;
+    }
+    
+    public List<OrderItem> getItemsToOrder() {
+        return itemsToOrder;
+    }
+    
+    public void setItemsToOrder(List<OrderItem> itemsToOrder) {
+        this.itemsToOrder = itemsToOrder;
+    }
+    
+    public boolean isFinished() {
+        return finished;
+    }
+    
+    public void setFinished(boolean finished) {
+        boolean actuallyFinished = getItemsToOrder().stream()
```

Comment:
Degene die de methode oproept weet niet of het nu finished is of niet.
Indien er extra logica is voorzien moet er of wel javadoc zijn die het uitlegt of wel moet de naam van de methode duidelijk zijn.

Reply:
Zou nog een exception moeten gooien als het niet lukt.

Reply:
Code adjusted.